scripts/exact-example.jl

```diff
@@ -0,0 +1,237 @@
+# # Example 4: Exact diagonalisation
+
+# When working with smaller systems, or when multiple eigenvalues of a system are required,
+# it's better to work with an exact diagonalization method. There are a few ways to go about
+# this, each with its pros and cons. The purpose of this tutorial is to show off the methods
+# as well as provide a few tips regarding them.
+
+# A runnable script for this example is located
+# [here](https://github.com/joachimbrand/Rimu.jl/blob/develop/scripts/exact-example.jl).
+# Run it with `julia exact-example.jl`.
+
+# We start by loading the required modules.
+
+using Rimu
+using LinearAlgebra # eigen, eigvals, and eigvecs
+using Arpack        # eigs
+using KrylovKit     # eigsolve
+using Plots
+using LaTeXStrings
+gr() # hide
+nothing # hide
+
+# ## Preliminaries
+
+# In this example, we will look at a small example, a three-fermion Hamiltonian in one
+# dimension, formulated in momentum space. To start, we define some convenience functions.
+
+# In Rimu, momentum Fock states are stored such that the zero momentum state has a particle
+# in the middle of the address.
+
+# Below is a function that constructs a Fock state with total momentum 1 and ``M`` momentum
+# modes.
+
+function init_address(M)
+    return FermiFS2C(
+        [i == cld(M, 2) || i == cld(M, 2) + 1 ? 1 : 0 for i in 1:M],
+        [i == cld(M, 2) ? 1 : 0 for i in 1:M],
+    )
+end
+nothing # hide
+
+# The Hamiltonians in Rimu are dimensionless. We use the following function to convert the
+# units to physical units (``\frac{\hbar^2}{mL}``). This function also handles
```

Comment:
I think you want to say which units, rather than "physical units", and \hbar does not come into the function below, which is a little confusing.

Generally I don't like to say that a Hamiltonian (or any physical quantity) is dimensionless, but instead refer to a reference energy.  In this case it would be an energy constructed from Hubbard model parameters.

Reply:
I am also not sure if this example even needs to use realistic parameters, which would require a proper explanation of how they relte to the Hubbard model. The examples could just use some arbitrarily chosen values.